app/controllers/integrations/github_controller.rb

```diff
@@ -11,8 +11,21 @@ def self.secret_token
     ENV['GITHUB_HOOK_SECRET']
   end
 
+  def create
+    if github_event_type == "status"
+      handle_commit_status_event
+    else
+      super
+    end
+  end
+
   protected
 
+  def handle_commit_status_event
```

Comment:
The thing is – it doesn't expire the caches. It merely marks the release as being updated. The fact that our caching system uses that information is incidental – it could be used in other contexts as well. That provides a nice level of decoupling IMO.